### DISCUSSION TITLE
ANTLR4 [@types/antlr4] types seems incomplete (and a little hard to use)

### DISCUSSION THREAD

**Discussion by JoeNemo** [DefinitelyTyped/DefinitelyTyped#61897] [Issues with a @types package]
Maybe everybody else who uses this type library is totally Ok with everything.   I don't want to presume how other people use tsconfig and package.json to get things to work for them.   However, I am trying to use antlr in a "pure" ESM module context (NOT CommonJS/require) with a single import like:

`import antlr from "antlr4";`

Doing this does not give access to antlr.tree or antlr.error.    I can make other TS imports form error and tree, but that doesn't generate nice JS behavior.    The resultant import in java is going to be exactly "import antlr from 'antlr4';' anyway, and the antlr module loaded by JS has antlr.error and antlr.tree in it.  So why would I need to have multiple imports at the TS level.

So I added some re-exports to the index.d.ts and added a few missing function/method declarations, and now I am a happy antlr4 in TS user.    I would make a PR if my troubles are not due to misuse.

[a comment by an automated account is not shown]

**Comment by Codex-** [comment]
Hi Joe, i've done a big overhaul of the types: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/62582

As for sensible exports from the Antlr runtime package itself:
- https://github.com/antlr/antlr4/issues/3912
- https://github.com/antlr/antlr4/pull/3913
